crossplane/builder.py

```diff
@@ -63,11 +63,30 @@ def _enquote(arg):
 def build(payload, indent=4, tabs=False):
     padding = '\t' if tabs else ' ' * indent
 
-    def _build_lines(objs, depth):
-        margin = padding * depth
+    def _put_line(line, obj):
+        prev_obj = build.prev_obj
```

Comment:
Although I'm sure this works, you shouldn't set attributes of a function. Instead, you can just initialize 
```python
prev_obj = None
depth = -1
```
 or whatever as local variables just above the `_put_line` definition. This would allow both variables to be within the scope of both `_put_line` and `_build_lines`, and both would get  deallocated after `build` finishes.